src/clj/chronograph/auth.clj

```diff
@@ -0,0 +1,39 @@
+(ns chronograph.auth
+  (:require [buddy.sign.jwt :as jwt]
+            [chronograph.config :as config]
+            [taoensso.timbre :as log]
+            [ring.util.response :as response])
+  (:import (java.time Instant)
+           (java.time.temporal ChronoUnit)))
+
+(defn- secret-key []
+  (get-in config/config [:auth :token-signing-key]))
+
+(defn create-token [id email name photo-url]
+  (jwt/sign {:id        id
+             :email     email
+             :name      name
+             :photo-url photo-url
+             :exp       (-> (Instant/now)
+                            (.plus (get-in config/config [:auth :token-expiry-in-seconds])
+                                   ChronoUnit/SECONDS)
+                            (.getEpochSecond))}
+            (secret-key)))
+
+(defn unsign-token [token]
+  (try
+    (jwt/unsign token (secret-key))
+    (catch Exception e
+      (log/error e "Unsigning a token failed")
+      nil)))
```

Comment:
Instead of returning nil, can we return something more meaningful here?

Reply:
We could, but at the moment there's no consumer that requires meaningful information, I think. We do not (and IMO should not) return a specific error to the API callers, for instance. They can just assume that they need a new token.

Reply:
The reason to return a meaningful to avoid implicit nil checks in the code, we need not return specific errors to user. 

Wondering if we should have a common strategy around error handling, maybe something like failjure?

Reply:
> avoid implicit nil checks in the code

Could you elaborate on this?

Using `nil` is convenient because of nil punning. It works well with built-in macros like `some->`, `if-let`, etc.

I am open to using something like `failjure`. I've used it in the past to good effect, but I think it requires buy-in from the team. Let's discuss more in person.